astropy/cosmology/core.py

```diff
@@ -85,15 +137,20 @@ class Cosmology(metaclass=abc.ABCMeta):
     override the ``_register_cls`` classmethod in the subclass.
     """
 
-    meta = MetaData()
+    if not PYTHON_LT_3_10:
+        _: KW_ONLY
```

Comment:
maybe I'm only asking because I don't have a clear idea of how this class already works but this looks like a breaking change, as well as a Python version-dependent behaviour (but that's secondary since 3.9 is going away soon anyway).
I'm worried that the breaking change would have been fine in 6.0 but not for 6.1.

Reply:
You're technically correct that this PR changes `name` and `meta` from positional-or-keyword arguments to keyword-only arguments, which is a change. However in practice this is not a change. I couldn't find a single example in Astropy, on GitHub, nor GitLab where anyone ever passes these arguments positionally (the closest is using `inspect.BoundArguments()` which auto-splits args and kwargs, but this is not affected by the change for that same reason). There's a good reason no one positionally passes `name` and `meta`: in all the non-abstract classes `name` and `meta` appear at the end of the `__init__` signature. That, and since different classes have different number of preceding arguments, in practice these have always been keyword arguments. As such I think it's fine for v6.1.

Reply:
As for the Python-version dependent behavior, this is actually why this PR has to wait for Py3.10. Earlier forms of dataclasses don't understand how to rearrange non-defaulted parameters based on redefinitions when inheriting from a class with those same parameters. Ie you can't redefine `name` and `meta` in a subclass to put them at the end if you have a parameter `H0` that doesn't have a default value.